### PR TITLE
Add filter for GCP Org Policies

### PR DESCRIPTION
**Pull request by jkreider-mckinsey**

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/resourcemanager.py
Comment (ajkerrigan) [inline tools/c7n_gcp/c7n_gcp/resources/resourcemanager.py:415]: This is a clever idea - extending `ListItemFilter` rather than `ValueFilter` for custom filters that expect to return a list :bulb:

**Review by ajkerrigan — COMMENTED** [review]
This looks good to me, thanks for the contribution! I'd like a second opinion from someone with better-developed gcp instincts before merging.

Specifically, I can see not including `key` in the filter schema because we know up front what it needs to look at. But is there a case to support the `count` and `count_op` parameters from `ListItemFilter`? I'm not sure...

**Comment by kapilt** [comment]
thanks for the pr, I was actually thinking we should have these as their own resource, they get tracked directly in cloud asset inventory as their own type.

**Comment by kapilt** [comment]
it does seem a bit harder to model as a resource due to the need factor in variable scope (org, folder, project) when fetching.

**Comment by kapilt** [comment]
actually we could do it as a child resource manager with scope as parent, would work but that would require multiplying out org-policy, folder-policy, project-policy as resources. alternatively we could have a policy filter on each.

**Review by kapilt — APPROVED** [review]
thanks, lgtm

**Comment by kapilt** [comment]
Closing in favor #8982

**Pull request closed without merging**
